.github/workflows/ci.yaml

```diff
@@ -0,0 +1,188 @@
+name: ci
+
+on:
+  push:
+    branches: ['*']
+    tags:
+      - v*
+  pull_request:
+    type: [opened, reopened, edited]
+  schedule:
+    # run every night at midnight
+    - cron:  '0 0 * * *'
+
+jobs:
```

Comment:
I think it would also be good to utilize this action to skip redundant builds - https://github.com/fkirc/skip-duplicate-actions.

I've been using it quite a lot recently with various GHA workflows. In a sense, it's similar to Circle CI auto cancel old workflows, but in addition to that, it also allows you to skip a build if a build for a specific SHA and set of files already exist.

Having said that, we should just apply that check to non master and version branches. Something along the lines of https://github.com/apache/libcloud/blob/trunk/.github/workflows/main.yml#L31.